### PR TITLE
Proxy: use a global switch for Filter Middleware

### PR DESCRIPTION
**Pull request by marwan-at-work**
Fixes last part of https://github.com/gomods/athens/issues/495

### CODE REVIEW

File: pkg/module/filter.go [hunk 1]
```diff
@@ -118,6 +119,7 @@ func (f *Filter) initFromConfig() {
 	lines, err := getConfigLines()
 
 	if err != nil || len(lines) == 0 {
+		f.Off = true
```
Comment (michalpristas) [inline pkg/module/filter.go]: can we also have a ENV override? what do you think? even with a filter file provided, we could turn it off

Comment (marwan-at-work) [reply pkg/module/filter.go]: @michalpristas I like that. I'll add one. I think the Filter should be Off by default unless provided with an actual configuration (or ENV override). But once we have Olympus working well, then maybe the Filter should be On by default unless the user explicitly turns it off by ENV only.

Comment (marpio) [reply pkg/module/filter.go]: Could we also skip the whole middleware if the environment var is set?

Comment (marwan-at-work) [reply pkg/module/filter.go]: @michalpristas i liked your ENV idea more than my "Off" idea that I basically removed my code 😄 -- it's much easier to just not use the middleware at all if we want it off.

File: pkg/config/env/filter.go [hunk 2]
```diff
@@ -9,3 +9,11 @@ const defaultConfigurationFileName = "filter.conf"
 func FilterConfigurationFileName() string {
 	return envy.Get("ATHENS_FILTER_FILENAME", defaultConfigurationFileName)
 }
+
+// FilterOff checks PROXY_FILTER_OFF env and returns
+// true of it's equal to "true", otherwise false always.
+// It defaults to "true" until Olympus is the default
+// place to grab modules before the Proxy.
+func FilterOff() bool {
+	return envy.Get("PROXY_FILTER_OFF", "true") == "true"
```
Comment (robjloranger) [inline pkg/config/env/filter.go]: Should the envy call default to `false` as the check is whether the filter is off?

like `envy.Get("PROXY_FILTER_OFF", "false") == "true"`
if `PROXY_FILTER_OFF` was `true`, result would be `true`
if `PROXY_FILTER_OFF` was 'false' result would be `false`

the way it is now, with default of true, if the user does not set `PROXY_FILTER_OFF` the result will be `true`.

Comment (robjloranger) [reply pkg/config/env/filter.go]: I guess not, re-reading the comment clears that up. :D

Comment (marwan-at-work) [reply pkg/config/env/filter.go]: Yes by default, it's just the Proxy, no filtering needed. But we will use Olympus by default once Olympus is ready.

**Comment by marwan-at-work** [comment]
@michalpristas merging this to unblock myself and @carolynvs -- feel free to make comments if you have any and I'll open another PR :)

**Pull request merged**